### DISCUSSION TITLE
[dev-alloc] Removing `bitsPerElement` parameter from alloc APIs.

### DISCUSSION THREAD

**Discussion by fasiddique** [UVA-LavaLab/PIMeval-PIMbench#131] [PIMeval]
I am wondering do we actually need the `bitsPerElement` parameter in the allocation APIs. Can we not just derive it from the data type?

**Comment by deyuan** [comment]
For functional simulation, we can derive bitsPerElement based on data types. The APIs can be simplified.
For micro-ops level simulation, it would be good to keep to flexibility for advanced usage.

**Comment by deyuan** [comment]
Closed with PR #168
